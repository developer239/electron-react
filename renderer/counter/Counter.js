import React, { Component } from 'react';
import styled from 'styled-components';


const Container = styled.div`
  padding: 20px 20px
`;

const Button = styled.button`
  width: 100%;
  margin: 10px 0;
`;

export default class Main extends Component {
  render() {
    return (
      <Container>
        <Button>Increment</Button>
        <br />
        <Button>Decrement</Button>
      </Container>
    );
  }
}
