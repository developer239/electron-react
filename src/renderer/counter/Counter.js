import React from 'react'
import PropTypes from 'prop-types'
import { compose, withState, withHandlers } from 'recompose'
import { COUNTER_INCREMENT, COUNTER_DECREMENT, COUNTER_SET_VALUE } from '../../_shared/constants'
import { send } from '../_shared/messageHelper'
import {
  Container,
  Content,
  Button,
  Input,
} from '../_shared/components'


const Counter = ({
  handleIncrement,
  handleDecrement,
  handleClearValue,
  inputValue,
  handleInputChange,
  handleSetDirectly,
}) => (
  (
    <Container>
      <Content>
        <Button onClick={handleIncrement}>Increment</Button>
        <Button onClick={handleDecrement}>Decrement</Button>
        <Button onClick={handleClearValue}>Clear</Button>
        Set Directly:
        <Input
          type="number"
          value={inputValue}
          onChange={handleInputChange}
        /> <br />
        <Button onClick={handleSetDirectly}>Save</Button>
      </Content>
    </Container>
  )
)

Counter.propTypes = {
  handleIncrement: PropTypes.func.isRequired,
  handleDecrement: PropTypes.func.isRequired,
  handleClearValue: PropTypes.func.isRequired,
  inputValue: PropTypes.number.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  handleSetDirectly: PropTypes.func.isRequired,
}

const enhance = compose(
  withState('inputValue', 'setInputValue', 0),
  withHandlers({
    handleIncrement: () => () => send(COUNTER_INCREMENT),
    handleDecrement: () => () => send(COUNTER_DECREMENT),
    handleSetDirectly: ({ inputValue, setInputValue }) => () => {
      send(COUNTER_SET_VALUE, inputValue)
      setInputValue(0)
    },
    handleInputChange: ({ setInputValue }) => event =>
      setInputValue(parseInt(event.target.value, 10)),
    handleClearValue: ({ setInputValue }) => () => {
      send(COUNTER_SET_VALUE, 0)
      setInputValue(0)
    },
  }),
)

export default enhance(Counter)
