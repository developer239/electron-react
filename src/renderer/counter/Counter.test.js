import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import { Counter } from './Counter'


describe('Counter Component', () => {
  const handleIncrement = sinon.fake()
  const handleDecrement = sinon.fake()
  const handleClearValue = sinon.fake()
  const inputValue = 0
  const handleInputChange = sinon.fake()
  const handleSetDirectly = sinon.fake()

  it('renders with all props', () => {
    const wrapper = shallow((
      <Counter
        handleIncrement={handleIncrement}
        handleDecrement={handleDecrement}
        handleClearValue={handleClearValue}
        inputValue={inputValue}
        handleInputChange={handleInputChange}
        handleSetDirectly={handleSetDirectly}
      />
    ))

    expect(wrapper).toMatchSnapshot()
  })

  it('it calls counter handlers', () => {
    const wrapper = shallow((
      <Counter
        handleIncrement={handleIncrement}
        handleDecrement={handleDecrement}
        handleClearValue={handleClearValue}
        inputValue={inputValue}
        handleInputChange={handleInputChange}
        handleSetDirectly={handleSetDirectly}
      />
    ))
    const buttons = wrapper.find('Button')

    // Increment Handler
    buttons.at(0).simulate('click')
    expect(handleIncrement.callCount).toBe(1)

    // Decrement Handler
    buttons.at(1).simulate('click')
    expect(handleDecrement.callCount).toBe(1)

    // Clear Handler
    buttons.at(2).simulate('click')
    expect(handleClearValue.callCount).toBe(1)

    // Input Change Handler
    wrapper.find('Input').simulate('change', 1)
    expect(handleInputChange.callCount).toBe(1)

    // Handle Set Directly Button
    buttons.at(3).simulate('click')
    expect(handleSetDirectly.callCount).toBe(1)
  })
})
