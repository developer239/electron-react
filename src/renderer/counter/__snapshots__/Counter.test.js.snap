// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    handleClearValue={[Function]}
    handleDecrement={[Function]}
    handleIncrement={[Function]}
    handleInputChange={[Function]}
    handleSetDirectly={[Function]}
    inputValue={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Content>
        <Button
          onClick={[Function]}
        >
          Increment
        </Button>
        <Button
          onClick={[Function]}
        >
          Decrement
        </Button>
        <Button
          onClick={[Function]}
        >
          Clear
        </Button>
        Set Directly:
        <Input
          onChange={[Function]}
          type="number"
          value={0}
        />
         
        <br />
        <Button
          onClick={[Function]}
        >
          Save
        </Button>
      </Content>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Button
            onClick={[Function]}
          >
            Increment
          </Button>,
          <Button
            onClick={[Function]}
          >
            Decrement
          </Button>,
          <Button
            onClick={[Function]}
          >
            Clear
          </Button>,
          "Set Directly:",
          <Input
            onChange={[Function]}
            type="number"
            value={0}
          />,
          " ",
          <br />,
          <Button
            onClick={[Function]}
          >
            Save
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Increment",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increment",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Decrement",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Decrement",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Clear",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clear",
          "type": [Function],
        },
        "Set Directly:",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "type": "number",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Save",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Content>
          <Button
            onClick={[Function]}
          >
            Increment
          </Button>
          <Button
            onClick={[Function]}
          >
            Decrement
          </Button>
          <Button
            onClick={[Function]}
          >
            Clear
          </Button>
          Set Directly:
          <Input
            onChange={[Function]}
            type="number"
            value={0}
          />
           
          <br />
          <Button
            onClick={[Function]}
          >
            Save
          </Button>
        </Content>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              onClick={[Function]}
            >
              Increment
            </Button>,
            <Button
              onClick={[Function]}
            >
              Decrement
            </Button>,
            <Button
              onClick={[Function]}
            >
              Clear
            </Button>,
            "Set Directly:",
            <Input
              onChange={[Function]}
              type="number"
              value={0}
            />,
            " ",
            <br />,
            <Button
              onClick={[Function]}
            >
              Save
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Increment",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Increment",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Decrement",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Decrement",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Clear",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Clear",
            "type": [Function],
          },
          "Set Directly:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "type": "number",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Save",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
